src/rust/engine/options/src/args.rs

```diff
@@ -129,15 +136,16 @@ impl Args {
                         Some(value.to_string())
                     },
                 });
-            } else if is_valid_scope_name(&arg_str) {
-                scope = Scope::Scope(arg_str)
+            } else if is_valid_scope_name(arg_str) {
+                scope = Scope::Scope(arg_str.to_string())
```

Comment:
We need the original arg_str for the arg_strs field, so we must copy it here. Ditto above.